lib/src/components/NxCheckbox/NxCheckbox.scss

```diff
@@ -21,24 +21,49 @@ $-width: 16px;
     @include nx-focus-border-helpers.radio-checkbox-focus-border($-height, $-width);
 
     border-radius: 4px;
-    left: -5px;
-    top: -5px;
+    left: -4px;
+    top: -4px;
   }
 
   .fa-check {
-    color: var(--nx-swatch-white);
+    color: var(--nx-color-component-background);
     font-size: var(--nx-font-size-s);
-    vertical-align: 4px;
+    vertical-align: 3px;
+    margin-left: 1px;
   }
 }
 
-.nx-checkbox__input:checked:disabled {
-  + .nx-checkbox__box {
-    background-color:var(--nx-color-interactive-background-disabled);
-    border-color: var(--nx-color-interactive-border-disabled);
+.nx-checkbox__input {
+  &:disabled {
+    &, &:hover {
+      + .nx-checkbox__box {
+        background-color: var(--nx-color-checkbox-disabled-background);
+        box-shadow: inset 0 0 0 var(--nx-radio-checkbox-border-width) var(--nx-color-checkbox-disabled-box-shadow);
+      }
+    }
+  }
+
+  &:checked {
+    + .nx-checkbox__box {
+      background-color: var(--nx-color-radio-checkbox-selected);
+      box-shadow: inset 0 0 0 var(--nx-radio-checkbox-border-width) var(--nx-color-radio-checkbox-selected);
+
+      &:hover {
+        box-shadow: inset 0 0 0 var(--nx-radio-checkbox-border-width) var(--nx-color-radio-checkbox-selected);
+      }
+    }
 
-    .fa-check {
-      color: var(--nx-color-text-disabled);
+    &:disabled {
+      &, &:hover {
+        + .nx-checkbox__box {
+          background-color: var(--nx-color-checkbox-disabled-background);
+          box-shadow: inset 0 0 0 var(--nx-radio-checkbox-border-width) var(--nx-color-checkbox-disabled-box-shadow);
+    
+          .fa-check {
+            color: var(--nx-color-checkbox-disabled-icon);
+          }
```

Comment:
The `background-color` and `box-shadow` here are redundant, and I believe the `.fa-check` block can be moved up to the disabled styles just above this without any harm.

Reply:
> The `background-color` and `box-shadow` here are redundant, and I believe the `.fa-check` block can be moved up to the disabled styles just above this without any harm.

I thought so too, but it is needed or else it doesn't show the disabled state when checked
Will investigate more.

<img width="226" alt="Screenshot 2023-04-11 at 9 21 35 AM" src="https://user-images.githubusercontent.com/10160030/231176096-e1f2cd47-2e56-4a6b-9932-68343847c1cb.png">

Reply:
> I thought so too, but it is needed or else it doesn't show the disabled state when checked
Will investigate more.

Try reversing the order of the `checked` and `disabled` blocks in the CSS.  They have equal specificity so disabled needs to come afterwards in order to win.

Reply:
> > I thought so too, but it is needed or else it doesn't show the disabled state when checked
> > Will investigate more.
> 
> Try reversing the order of the `checked` and `disabled` blocks in the CSS. They have equal specificity so disabled needs to come afterwards in order to win.

Thanks this worked for checkbox but not radio 🤔 I'm still playing around with it
c5f9fe75186d4edcbcfd8ad29c6ba6e4a53af5a0